sdk/js/src/User.ts

```diff
@@ -23,16 +23,18 @@ export class DVCPopulatedUser implements DVCUser {
     readonly platform: string
     readonly platformVersion: string
     readonly deviceModel: string
-    readonly sdkType: 'client' | 'server'
+    readonly sdkType: 'client'
     readonly sdkVersion: string
 
     constructor(user: DVCUser, options: DVCOptions, staticData?: StaticData, anonymousUserId?: string) {
-        if (!user.user_id && !user.isAnonymous) {
-            throw new Error('Must have a user_id, or have "isAnonymous" set on the user')
+        if (user.user_id?.trim() === '') {
+            throw new Error('A User cannot be created with a user_id that is an empty string')
         }
-
-        this.user_id =  user.isAnonymous ? anonymousUserId || uuidv4() : user.user_id || uuidv4()
-        this.isAnonymous = user.user_id ? false : (user.isAnonymous ?? true)
+        
+        this.user_id = (user.isAnonymous || !user.user_id) ? 
+            (anonymousUserId || uuidv4()) :
+            user.user_id
+        this.isAnonymous = (user.isAnonymous || !user.user_id)
```

Comment:
could simplify this, makes the conditional easier to read
```suggestion
        this.isAnonymous = (user.isAnonymous || !user.user_id)
        this.user_id = this.isAnonymous ? (anonymousUserId || uuidv4()) :user.user_id
```

Reply:
sadly, doesn't work due to the type inferencing... typescript will not allow this, as it can't tell that user.user_id would be defined... with the ternary in this line, it works because it checks of the existence in the `!user.user_id`